src/utils/merge.js

```diff
@@ -19,9 +22,40 @@ export default function mergeObjects (dest, ...sources) {
                 }
             }
             // Overwrite the previous destination value if the source property is: a scalar (number/string),
-            // an array, or a null value
+            // an array (unless using array merge operator below), or a null value
             else if (value !== undefined) {
-                dest[key] = value;
+                let merge_index = is_array && value.indexOf(array_merge_op); // special array merge syntax
```

Comment:
What is the `merge_index` if there are multiple occurrences of `array_merge_op` in the source array?

Reply:
It would be the first occurrence - what would you propose as desired behavior?

Reply:
First is good, just wanted to clarify how that case is handled.